### ISSUE TITLE
In-Game: [VM] Nested functions incorrectly share parameters

### ISSUE THREAD

**Issue by sanbox-irl** [YoYoGames/GameMaker-Bugs#4370]
### Description

```
function mapper(i) {
    show_debug_message("Who is running this code?");

    return 100;
}

function wrapper(input, second) {
    show_debug_message(string(second)); // this will print out the same thing as line 15 does

    return {
        input,
    }
}

show_debug_message("`mapper` = " + string(mapper));

var req = wrapper(
    array_map(["foo"]), // nb: `mapper` is not passed as a parameter to `array_map`!
    mapper // nb: no `()`, so no one is calling this function
);
show_debug_message(string(req)); // this array is `[100]`, seemingly because `mapper` was called
```

Running the above code should result in a compile time error where `array_map` needs a second parameter -- instead, it **uses the parameter in the above function**.

**Note: this is VM only and does not occur on YYC, though the error you get on the YYC here is very odd -- it says that "camera_create" doesn't have enough arguments, which is odd.**

### Expected Change

It should show a code error in-game.

### Steps To Reproduce

1. Start GameMaker
2. Run the above code.
3. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.11.1 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by YYDan** [comment]
It definitely should not "crash". Fixed your wording there ;)

**Comment by sanbox-irl** [comment]
@YYDan running array_map(["foo"]) should (and does) generate a crash stating a missing parameter. The compiler can catch it sometimes too and give a compile time warning but obviously you can alias the function enough that it can't tell. So yes, it should "crash". You can write "throw" if you want to be particular

**Comment by rwkay** [comment]
Fixed in 2024.2 - specifically for `array_map` it was not checking the minimum number of arguments that it needed
